### PR TITLE
🐛 Fix integration tests on EKS

### PR DESCRIPTION
**Pull request by czunker**

### CODE REVIEW

**Comment by imilchev** [comment]
I suppose this fixes #536

**Comment by czunker** [comment]
> I suppose this fixes #536

Yes, the one commit. I hope when I do not squash, that GitHub automatically closes it.

File: controllers/nodes/resources.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 const (
 	CronJobNameBase        = "-node-"
 	InventoryConfigMapBase = "-node-inventory-"
+	CronJobNameMaxLength   = 52
```
Comment (imilchev) [inline controllers/nodes/resources.go:24]: It is actually the resource name max length in k8s. It's not cronjob-specific. It's just that we use it for the cronjobs because we add the node names to the cronjob name and in some cases the node names can be very long

**Review by imilchev — COMMENTED** [review]
Are you going to rebase this PR on `main` or is that going to be done in another PR?

File: .github/workflows/release-tests.yaml [hunk 2]
```diff
@@ -46,31 +46,49 @@ jobs:
     strategy:
       matrix:
         k8s-version: ["1.23"]
-        k8s-distro: ["aws"]
+        k8s-distro: ["eks"]
```
Comment (imilchev) [inline .github/workflows/release-tests.yaml:49]: I think you don't need this. You can list just the `k8s-version` like I Did for AKS.

File: .github/workflows/release-tests.yaml [hunk 3]
```diff
@@ -46,31 +46,49 @@ jobs:
     strategy:
       matrix:
```
Comment (imilchev) [inline .github/workflows/release-tests.yaml:47]: It is important to add the `fail-fast: false` since if 1 of the k8s versions fails it will cancel all pending tests before the `terraform destroy`. That will leave dangling resources in AWS. I had that issue with AKS earlier today.

File: .github/workflows/release-tests.yaml [hunk 4]
```diff
@@ -83,12 +101,22 @@ jobs:
       - name: Run integration tests
         run: VERSION=${{ env.MONDOO_OPERATOR_IMAGE_TAG }} K8S_DISTRO=${{ matrix.k8s-distro }} make test/integration/ci
         env:
-          KUBE_CONFIG: ./github/terraform/${{ matrix.k8s-distro }}/kubeconfig
+          AWS_ACCESS_KEY_ID: ${{ github.event.inputs.AWS_ACCESS_KEY_ID }}
```
Comment (imilchev) [inline .github/workflows/release-tests.yaml:104]: why are these needed for the integration tests?

File: .github/workflows/release-tests.yaml [hunk 5]
```diff
@@ -83,12 +101,22 @@ jobs:
...
+          AWS_SECRET_ACCESS_KEY: ${{ github.event.inputs.AWS_SECRET_ACCESS_KEY }}
+          AWS_SESSION_TOKEN: ${{ github.event.inputs.AWS_SESSION_TOKEN }}
 
       - name: Clean up EKS terraform
-        run: terraform destroy
+        run: terraform destroy -auto-approve
+        env:
+          TF_VAR_test_name: ${{ env.MONDOO_OPERATOR_IMAGE_TAG }}
+          TF_VAR_kubernetes_version: ${{ matrix.k8s-version }}
+          AWS_ACCESS_KEY_ID: ${{ github.event.inputs.AWS_ACCESS_KEY_ID }}
```
Comment (imilchev) [inline .github/workflows/release-tests.yaml:113]: why not set these env vars at the job level instead of copy-pasting 5-6 lines for every step. Would make things easier to read

File: Makefile [hunk 6]
```diff
@@ -109,7 +109,7 @@ test/ci: manifests generate fmt vet envtest gotestsum
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) --arch=amd64 use $(ENVTEST_K8S_VERSION) -p path)" $(GOTESTSUM) --junitfile unit-tests.xml -- $(UNIT_TEST_PACKAGES) -coverprofile cover.out
 
 # Integration tests are run synchronously to avoid race conditions
-ifeq ($(K8S_DISTRO),aws)
+ifeq ($(K8S_DISTRO),eks)
```
Comment (imilchev) [inline Makefile:112]: I think on `main` it now says `aks`. Is there a way to do logical OR in this `ifeq`? I am a bit of a Makefile/bash noob

File: .github/workflows/release-tests.yaml (hunk 5 above)
Comment (czunker) [reply .github/workflows/release-tests.yaml:113]: Done.

File: .github/workflows/release-tests.yaml (hunk 3 above)
Comment (czunker) [reply .github/workflows/release-tests.yaml:47]: Thanks for the hint.

File: Makefile (hunk 6 above)
Comment (czunker) [reply Makefile:112]: I just looked it up. There is a construct you could use, but I do not think it is actually readable, so I used a more expressive solution.

File: controllers/nodes/resources.go (hunk 1 above)
Comment (czunker) [reply controllers/nodes/resources.go:24]: Moved that to a general package

**Comment by czunker** [comment]
> Are you going to rebase this PR on `main` or is that going to be done in another PR?
> 
> Nevermind, I saw there's a conflict now

I re-based on main, but I kept two separate files. The input fields are so different, that I do not think it makes sense to combine them.

File: .github/workflows/release-tests.yaml (hunk 2 above)
Comment (czunker) [reply .github/workflows/release-tests.yaml]: Deleted.

**Comment by czunker** [comment]
> > > Are you going to rebase this PR on `main` or is that going to be done in another PR?
> > > Nevermind, I saw there's a conflict now
> > 
> > 
> > I re-based on main, but I kept two separate files. The input fields are so different, that I do not think it makes sense to combine them.
> 
> I think the inputs are identical. Why don't you arrange the AWS creds to be added to the repo and remove the secrets input. Then we have just the operator and client versions as input. I think we should have 1 workflow for all clouds and then we have 1 matrix job for each provider

Missed the point, that secrets are separate.

Changed the files accordingly.

File: .github/workflows/cloud-tests.yaml [hunk 7]
```diff
@@ -116,10 +126,104 @@ jobs:
...
+    env:
+      TF_VAR_test_name: ${{ github.event.inputs.mondooOperatorImageTag }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_REGION: ${{ github.event.inputs.AWS_REGION }}
+
+    steps:
+      - name: hide values
+        run: |
+          echo "::add-mask::${{ github.event.inputs.AWS_ACCESS_KEY_ID }}"
```
Comment (imilchev) [inline .github/workflows/cloud-tests.yaml:149]: now that these are actual secrets I don't think this step is needed

File: .github/workflows/cloud-tests.yaml [hunk 8]
```diff
@@ -12,6 +12,11 @@ on:
         type: string
         default: main
         description: The image tag to use for the mondoo operator image
+      AWS_REGION:
```
Comment (imilchev) [inline .github/workflows/cloud-tests.yaml:15]: why not just hard-code this into the workflow. I wouldn't see why you would want it to be different

File: .github/workflows/cloud-tests.yaml (hunk 7 above)
Comment (czunker) [reply .github/workflows/cloud-tests.yaml]: Oh, they must have slipped back in during one of the rebases. Good catch. Thanks.

File: .github/workflows/cloud-tests.yaml (hunk 8 above)
Comment (czunker) [reply .github/workflows/cloud-tests.yaml]: During testing, I created them in the EU, but agree.

File: .github/workflows/cloud-tests.yaml [hunk 9]
```diff
@@ -116,10 +121,98 @@ jobs:
...
+        if: success() || failure()
+
+      - run: mv integration-tests.xml integration-tests-eks-${{ matrix.k8s-version }}.xml
+        if: success() || failure()
+
+      - name: Upload test results
+        uses: actions/upload-artifact@v3  # upload test results
+        if: success() || failure()        # run this step even if previous step failed
+        with:                             # upload a combined archive with unit and integration test results
+          name: test-results
```
[the remark this answers is not in the document]
Comment (czunker) [reply .github/workflows/cloud-tests.yaml]: Thanks. Fixed.

**Review by imilchev — APPROVED** [review]
